BiweeklyMeetings/2022-07-05.md

```diff
@@ -0,0 +1,274 @@
+
+**Matthias** - :  Yeah. So beyond that, I haven't gone through all the comments. We could go through the comments as part of this current biweekly as well, since we have a bit of time left, or we could leave that for another time and try to do again a summary of the conversation that had been happening. Any thoughts on that?
+
+**Sebastien** -:  Yeah. For this one, I feel like the summaries are nice, but I feel this one needs more than just a summary.
+
+**Matthias** - :  Of course. And as I said last time, I think this one can only move forward. At some point with regards the participation of IOG research. Without that, I'd say it's very difficult for that proposal to make concrete steps. And yeah, we have, I think two, yeah, Christopher and Colette. I think you are here both of you probably for the CIP 50, so maybe you have all sorts of stuff you want to discuss or say, please bring them up. Yeah, so I guess same situation for everyone. So I don't really have anything to add mainly in this call. We're talking about safety process in general, okay. And okay, so both of you. So is there any other items then we could discuss today before we wrap it up on maybe CIP 58? I don't think we have talked about that one yet. It was proposed and then proposed again in a new format, I think. So it's not from the same author, but it's from the same team as far as I understood, so it's superseding the previous proposal. So the first proposal was... Where it is? I'd lost it now.
+
+**Matthias** - :  Okay, it was PR number 268 and the second proposal was PR 283, which is a new Plutus set of built-ins about doing Bitwise operations in Plutus. So manipulations on bites, which is quite common to do on low level stuff. It's been missing from Plutus so far, there is very pro-support from manipulating bite string in general and doing Bitwise operations in Plutus. Although it can be quite interesting because of the, well, performance aspect it can also bring in the picture. So it's a bit risky these things usually. So I haven't gone through the proposal yet, but I hope that it's a balance reducing a somewhat safe API for doing Bitwise operation and not allow people to shoot themselves in the foot with this stuff. I see it's proposed by the mLab's team and it has also been discussed with the Plutus team quite extensively, it seems. So this doesn't come out of the blue basically. So I think it'll be good to have it for review next time so that we get some good time to go through it, have a few comments and re-bring the spotlight to it on the next biweekly.
```

Comment:
```suggestion
**Matthias** - :  Okay, it was PR number 268 and the second proposal was PR 283, which is a new Plutus set of built-ins about doing bitwise operations in Plutus. So manipulations on bits, which is quite common to do on low level stuff. It's been missing from Plutus so far, there is very no support from manipulating byte strings in general and doing bitwise operations in Plutus. Although it can be quite interesting because of the, well, performance aspect it can also bring in the picture. So it's a bit risky these things usually. So I haven't gone through the proposal yet, but I hope that it's introducing a somewhat safe API for doing bitwise operations and not allow people to shoot themselves in the foot with this stuff. I see it's proposed by the MLab's team and it has also been discussed with the Plutus team quite extensively, it seems. So this doesn't come out of the blue basically. So I think it'll be good to have it for review next time so that we get some good time to go through it, have a few comments and re-bring the spotlight to it on the next biweekly.
```